www/localPackage.ts

```diff
@@ -118,13 +117,18 @@ class LocalPackage extends Package implements ILocalPackage {
                     } else {
                         var invokeSuccessAndInstall = () => {
                             CodePushUtil.logMessage("Install succeeded.");
-                            installSuccess && installSuccess();
-                            /* no neeed for callbacks, the javascript context will reload */
-                            cordova.exec(() => { }, () => { }, "CodePush", "install", [deployDir.fullPath, installOptions.rollbackTimeout.toString(), installOptions.installMode.toString()]);
+                            if (installOptions.installMode === InstallMode.IMMEDIATE) {
+                                /* invoke success before navigating */
+                                installSuccess && installSuccess();
+                                /* no neeed for callbacks, the javascript context will reload */
+                                cordova.exec(() => { }, () => { }, "CodePush", "install", [deployDir.fullPath, installOptions.installMode.toString()]);
+                            } else {
+                                cordova.exec(() => { installSuccess && installSuccess(); }, () => { installError && installError(); }, "CodePush", "install", [deployDir.fullPath, installOptions.installMode.toString()]);
+                            }
                         };
 
                         var preInstallSuccess = () => {
-                            Sdk.reportStatus(AcquisitionStatus.DeploymentSucceeded);
+                            Sdk.reportStatus(AcquisitionStatus.DeploymentSucceeded, this);
```

Comment:
@lostintangent @geof90 @dtivel I am working on making this change but I think we need to distinguish between a failed install and a rollback. In case of an error, the developer would like to have more details about what happened.